[Phua Li Ting] iP

phualiting opened the pull request:
### DukePro
> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is, 

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

phualiting committed:
Add greet, echo, exit

phualiting committed:
Add add, list

phualiting committed:
Add mark as done, unmark

phualiting committed:
Add todos, events, deadlines

phualiting committed:
Add automated text ui testing

phualiting committed:
Add handle errors

phualiting committed:
Add delete

phualiting committed:
Add use enums

phualiting committed:
Add save

phualiting committed:
Add dates and times

phualiting committed:
Merge branch 'branch-Level-7'

phualiting committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

phualiting committed:
Fix date bug

phualiting committed:
Add more oop

phualiting committed:
Add organize into packages

phualiting committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

phualiting committed:
Add JUnit tests

phualiting committed:
Add checkstyle

phualiting committed:
Add Javadoc comments

phualiting committed:
Follow coding standard

phualiting committed:
Add find

phualiting committed:
Merge branch 'branch-A-JavaDoc'

phualiting committed:
Merge branch 'branch-A-CodingStandard'

phualiting committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

sprintaway commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+
+import java.util.Scanner;
+
+public class Duke {
```
Perhaps can add javadoc for Duke class?

sprintaway commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+        this.tasks = new TaskList(this.storage.loadTasks());
+    }
+
+    public enum Keyword {
```
Good usage of enum

sprintaway commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+                        this.tasks.add(new Todo(description, false));
+                    } catch (DukeException e) {
+                        this.ui.showError(e.getMessage());
+                    }
```
Would it be better to create a createToDo method in Parser similar to createDeadline and createEvent?

sprintaway commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,59 @@
+public class Parser {
+    private static String[] keywords = {"bye", "list", "mark", "unmark", "todo", "deadline", "event", "delete", "find"};
+
+    public Parser() {};
```
The coding standard doesn't say about empty constructors so I'm not sure but I think it's 
```suggestion
    public Parser() {
    }
```
I think the semicolon is not needed as well.

sprintaway commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,59 @@
+     * @return Event object.
+     */
+    public Event createEvent(String nextLine) {
+        String[]sections = nextLine.split(" /at ");
```
There should be a space before sections. Maybe consider changing sections to section for consistency with Line 41, createDeadline?

sprintaway commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+        this.ui.showTaskAdded(task);
+        this.list.add(task);
+        this.count++;
+        printTaskCount();
```
Perhaps printTaskCount() should be in the Ui class instead of the TaskList class?

sprintaway commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,69 @@
+ * Deals with interactions with the user.
+ */
+public class Ui {
+    public Ui() {};
```
Similar to my comment for Parser class

sprintaway commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,13 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
import static should come before normal imports (at least according to checkstyle.xml) 
The order rule "STATIC###STANDARD_JAVA_PACKAGE###SPECIAL_IMPORTS###THIRD_PARTY_PACKAGE" consists of:
      1. STATIC: static imports
      2. STANDARD_JAVA_PACKAGE: standard java/javax imports

Benjamin-Sim replied:
Yes, static imports should come before non-static imports

sprintaway reviewed (commented):
Overall, LGTM. Well done 👍

phualiting committed:
Add GUI

phualiting committed:
Merge branch 'branch-Level-10'

phualiting committed:
Fix coding standard

Benjamin-Sim commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke;
+
+public class DukeException extends Exception {
```
Perhaps add a JavaDoc comment for the DukeException class?

Benjamin-Sim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,82 @@
+        return Ui.showWelcome();
+    }
+
+    public String getResponse(String input) {
```
Perhaps add a JavaDoc comment for the getResponse method?

Benjamin-Sim commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,19 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
```
Same as previous; Static imports should come before non-static imports

Benjamin-Sim commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,19 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Same as previous; Static imports should come before non-static imports

Benjamin-Sim reviewed (commented):
Overall, code is well formatted and follows the coding standard closely. Well done! The only minor issue is the static imports, but at least you were consistent in having the static imports come after the non-static imports in all your files.

One other suggestion is to organise your files into separate packages based on their functions, e.g. one package for Task classes, one package for GUI classes, etc.

phualiting committed:
Fix gui bug

phualiting committed:
Add assertions

Incorrect values of arguments for some methods can cause errors.

Assertions check for possible bugs in the code during runtime.

Let's include assertions to check values of variables at certain
points in the code to detect bugs.

phualiting committed:
Improve code quality

Code has sections that may be hard to understand.

Improving code quality would make the code clearer.

Let's make code more readable by making the happy path prominent.

phualiting committed:
Merge pull request #1 from phualiting/branch-A-Assertions

Add assertions

phualiting committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java

phualiting committed:
Merge pull request #2 from phualiting/branch-A-CodeQuality

Improve code quality

phualiting committed:
Add tagging

phualiting committed:
Merge branch 'branch-BCD-Extension'

phualiting committed:
Update images

phualiting committed:
Finalize code

phualiting committed:
Update README, add product screenshot

phualiting committed:
Update README
